### ISSUE TITLE
Give reflection-based row mappers a column name prefix option

### ISSUE THREAD

**Issue by qualidafial** [jdbi/jdbi#532]
Applies to: `BeanMapper`, `ConstructorMapper`, `FieldMapper`.

The specific use case I have in mind is with join row mappers:

``` java
class File {
  int id;
  String name;
  String contents;
}

class Directory {
  int id;
  String name;
  List<File> files;
}
```

I can't use reflection-based mappers with `JoinRowMapper` here, because the classes share the same property names. However if I could get a mapper that applies a column name prefix e.g. `f_` and `d_`, then I could exploit those prefixes to disambiguate those common property names:

Fluent:

``` java
handle.createQuery("select d.id d_id, d.name d_name, " +
    "f.id f_id, f.name f_name, f.contents f_contents " +
    "from directories d left join files f on d.id = f.directory_id")
    .registerRowMapper(BeanMapperFactory.forType("f_", File.class))
    .registerRowMapper(BeanMapperFactory.forType("d_", Directory.class))
    .map(JoinRowMapper.forTypes(Directory.class, File.class))
    .list();
```

SQL Object:

``` java
@RegisterBeanMapper(value={File.class,Directory.class}, prefix={"f_", "d_"})
@RegisterJoinRowMapper({File.class, Directory.class})
@SqlQuery("select d.id d_id, d.name d_name, " +
    "f.id f_id, f.name f_name, f.contents f_contents " +
    "from directories d left join files f on d.id = f.directory_id")
List<JoinRow> listFiles();
```

**Comment by qualidafial** [comment]
Fix in #552

**Comment by qualidafial** [comment]
Fixed in #552
